content/appendices.md

```diff
@@ -611,11 +611,20 @@ characters permitted in user ID localparts. There are currently active
 users whose user IDs do not conform to the permitted character set, and
 a number of rooms whose history includes events with a `sender` which
 does not conform. In order to handle these rooms successfully, clients
-and servers MUST accept user IDs with localparts from the expanded
-character set:
+and servers MUST accept user IDs with localparts consisting of any legal
+unicode codepoint except for `:`, including zero characters. Localparts
+MUST be valid UTF-8 sequences.
+
+Servers SHOULD NOT produce user IDs with localparts outside of the following
+character set, and SHOULD NOT forward such user IDs to clients when referenced
+outside the context of an event. For example, device list updates from "invalid"
+user IDs would be dropped by the receiving server.
 
     extended_user_id_char = %x21-39 / %x3B-7E  ; all ASCII printing chars except :
 
+A future room version may prevent users using a historical character set
+from participating. Use of the historical character set is *deprecated*.
```

Comment:
I'm still not entirely following this.

As far as I can tell, we are suggesting that there are three different classes:

1. "Conformant" user IDs, consisting only of `a-z`, `0-9`, `.`, `_`, `=`, `-`, `/`, and `+`, which we'd like servers to generate.
2. "Historical" user IDs set 1, consisting of any printing ASCII except `:`, which we have to treat the same as "Conformant" user IDs.
3. "Historical" user IDs set 2, consisting of any unicode codepoints except `:`, which we filter out. Sometimes.

I'm still not entirely following the logic that says we need three different classes: do we really?

If we're going to go with it, can we try to make it clearer that we have them? Maybe give each of them their own name rather than smashing two of them together as "historical". It reads rather confusingly at the moment.

Reply:
If things like dropping device list updates from users in set 2 is done, then it must be distinct from set 1, because set 1 contains real users. Some appservices still create uppercase users too (e.g. matrix-appservice-irc: https://github.com/matrix-org/matrix-appservice-irc/issues/1399).

Other than that I guess they don't really need to be separate.

Reply:
@turt2live this is the main outstanding thing that I'm not sure about

Reply:
Right, I get this now. The proposal is to define 3 different classes of user IDs:

1. "Conformant" user IDs, consisting only of `a-z`, `0-9`, `.`, `_`, `=`, `-`, `/`, and `+`. Servers SHOULD generate user-ids using only this set.
2. "Historical" user IDs, which contain characters outside the above set, but still in the range `%x21-39 / %x3B-7E` (ie, printing ASCII). There are real users in this set, so we have to treat them the same as "Conformant" user IDs, but servers SHOULD NOT generate such user IDs.
3. "Non-conformant" user IDs, which contain characters outside the printing ASCII range. Servers must accept these in event fields such as `sender` to avoid breaking the fabric of the room, but can otherwise ignore them, for example not sending device-list updates for such users.